src/python/lib/mesos/http.py

```diff
@@ -20,45 +20,46 @@
 Classes and functions for interacting with the Mesos HTTP restful API
 """
 
-from __future__ import absolute_import
-
+import copy
 import logging
-import urlparse
-
-# requests must be imported after all native packages
 import requests
 import tenacity
 import ujson
+import urlparse
 
-from mesos.exceptions import (HTTPException, MesosAuthenticationException,
-                              MesosAuthorizationException, MesosBadRequest,
-                              MesosUnprocessableException)
+from mesos.exceptions import HTTPException
+from mesos.exceptions import MesosException
 
-METHOD_HEAD = 'HEAD'
-METHOD_GET = 'GET'
-METHOD_POST = 'POST'
-METHOD_PUT = 'PUT'
-METHOD_PATCH = 'PATCH'
-METHOD_DELETE = 'DELETE'
+METHODS = ["HEAD", "GET", "POST", "PUT", "PATCH", "DELETE"]
 
 REQUEST_JSON_HEADERS = {'Accept': 'application/json'}
 REQUEST_GZIP_HEADERS = {'Accept-Encoding': 'gzip'}
 
-LOGGER = logging.getLogger(__name__)
+BASE_HEADERS = {}
+DEFAULT_TIMEOUT = 30
+DEFAULT_AUTH = None
+DEFAULT_USE_GZIP_ENCODING = None
+DEFAULT_MAX_ATTEMPTS = 3
 
 
-def simple_url_join(base_url, other_url):
+def simple_urljoin(base, other):
     """
-    Do a join by rstrip'ing / from base_url and lstrp'ing / from other_url.
+    Do a join by rstrip'ing '/' from base and lstrp'ing '/' from other.
 
-    This is needed since urlparse.urljoin tries to be too smart
-    and strips the subpath from base_url.
+    This function is needed since 'urlparse.urljoin' tries to be too
+    smart and strips the subpath from the base.
 
-    :type base_url: str
-    :type other_url: str
+    :type base: str
+    :type other: str
     :rtype: str
     """
-    return '/'.join([base_url.rstrip('/'), other_url.lstrip('/')])
+    try:
+        url = '/'.join([base.rstrip('/'), other.lstrip('/')])
+    except Exception as exception:
```

Comment:
catching all exceptions here essentially reduces the information content of any exceptions raised within the try block.  Example: `None` was accidentally passed for the `other ` param, but instead of getting an `AttributeError` and an informative stack trace that points all the way to the origin of the exception, we get MesosException instead with only `'NoneType' object has no attribute 'rstrip'` in the message. This is a bit of a contrived example, but best practice for Python exceptions is to only handle exceptions that you know how to handle, and letting unexpected exceptions expose bugs that need to be fixed.